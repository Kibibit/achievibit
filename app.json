{
  "name": "achievibit",
  "description": "Github Gamification - Achievements system as a GitHub WebHook. http://achievibit.kibibit.io/",
  "scripts": {
  },
  "env": {
  },
  "formation": {
  },
  "addons": [

  ],
  "buildpacks": [
    {
      "url": "heroku/nodejs"
    }
  ]
}
