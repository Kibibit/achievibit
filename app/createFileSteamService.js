var gulp = require('gulp'),
    gutil = require('gulp-util');

var createFileStreamService = {};

createFileStreamService.create = function string_src(filename, string) {
  var src = require('stream').Readable({ objectMode: true })
  src._read = function () {
    this.push(new gutil.File({
      cwd: '',
      base: '',
      path: filename,
      contents: new Buffer(string)
    }))
    this.push(null)
  }
  return src
};

module.exports = createFileStreamService;
