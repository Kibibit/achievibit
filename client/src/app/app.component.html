<mat-toolbar color="primary">
           <span id="site-name">kibibit Client Side</span>
</mat-toolbar>

<div style="display: flex; align-items: center; justify-content: center;">
  <h1 class="mat-display-1">Start Writing your app!</h1>

</div>

<router-outlet></router-outlet>