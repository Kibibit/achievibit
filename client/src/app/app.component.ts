import { Component } from '@angular/core';
import { webConsolelogo } from '@kibibit/consologo';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  title = 'kibibit client';

  constructor() {
    webConsolelogo('kibibit client template', [
      'kibibit server-client template',
      'change this up in app.component.ts'
    ]);
  }
}
