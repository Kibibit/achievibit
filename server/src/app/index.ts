/**
 * @file Automatically generated by barrelsby.
 */

export * from './app.controller';
export * from './app.module';
