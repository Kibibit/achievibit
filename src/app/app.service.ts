import { Injectable } from '@nestjs/common';

import { ConfigService } from '@kb-config';
import { PackageDetailsDto } from '@kb-models';

@Injectable()
export class AppService {
  constructor(private configService: ConfigService) { }
  getPackageDetails(): PackageDetailsDto {

    return this.configService.packageDetails;
  }
}
