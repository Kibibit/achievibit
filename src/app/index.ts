/**
 * @file Automatically generated by barrelsby.
 */

export * from './app.controller';
export * from './app.module';
export * from './app.service';
