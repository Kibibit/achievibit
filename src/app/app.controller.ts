import {
  Body,
  ClassSerializerInterceptor,
  Controller,
  Get,
  Headers,
  HttpStatus,
  Post,
  Render,
  UseInterceptors,
} from '@nestjs/common';
import { ApiOperation, ApiResponse } from '@nestjs/swagger';
import { groupBy } from 'lodash';

import { PackageDetailsDto } from '@kb-models';
import { GithubEventManagerService, ReposService, UsersService } from '@kb-modules';

import { AppService } from './app.service';

@Controller()
export class AppController {
  constructor(
    private readonly appService: AppService,
    private readonly reposService: ReposService,
    private readonly usersService: UsersService,
    private readonly githubEventManagerService: GithubEventManagerService
  ) { }

  @Get()
  @ApiOperation({ summary: `Return achievibit's homepage` })
  @Render('index.njk')
  async getHomepage(): Promise<any> {
    const repos = await this.reposService.findAll();
    const allUsers = await this.usersService.findAll();
    const groupedUsers = groupBy(allUsers, 'organization');
    const users = groupedUsers.undefined;
    const organizations = groupedUsers.true;

    return { repos, users, organizations };
  }

  @Get('api')
  @ApiOperation({ summary: `API Hello endpoint. Returns achievebit's package information` })
  @ApiResponse({
    status: HttpStatus.OK,
    description: `achievibit's package information`,
    type: PackageDetailsDto
  })
  @UseInterceptors(ClassSerializerInterceptor)
  async getAchievibitPackageInfo(): Promise<PackageDetailsDto> {
    return await this.appService.getPackageDetails();
  }

  @Post()
  @ApiOperation({ summary: `Recieve GitHub Webhooks` })
  async recieveGitHubWebhooks(@Headers('x-github-event') githubEvent: string, @Body() eventBody: any) {
    await this.githubEventManagerService.postFromWebhook(githubEvent, eventBody);

    return;
  }
}
