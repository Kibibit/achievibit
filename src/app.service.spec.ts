import { Test, TestingModule } from '@nestjs/testing';
import { classToPlain } from 'class-transformer';

import { AppService } from './app.service';

describe('AppService', () => {
  let service: AppService;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [ AppService ]
    }).compile();

    service = module.get<AppService>(AppService);
  });

  it('should be defined', async () => {
    expect(service).toBeDefined();
  });

  it('should return package info', async () => {
    const packageDetails = classToPlain(await service.getPackageDetails());

    expect(packageDetails).toMatchSnapshot();
  });

  it('should re-use the same promise on different calls', async () => {
    const packageDetailsPromiseCall1 = service.getPackageDetails();
    const packageDetailsPromiseCall2 = service.getPackageDetails();

    expect(packageDetailsPromiseCall1).toEqual(packageDetailsPromiseCall2);
  });
});
