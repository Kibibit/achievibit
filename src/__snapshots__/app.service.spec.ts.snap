// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppService should return package info 1`] = `
Object {
  "author": "",
  "browserslist": undefined,
  "bugs": undefined,
  "contributors": undefined,
  "description": "",
  "engines": undefined,
  "homepage": undefined,
  "keywords": undefined,
  "license": "MIT",
  "name": "achievibit",
  "repository": undefined,
  "version": "0.0.1",
}
`;
